src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -62,6 +89,13 @@
      * @return string with unique names, like "Distinct names: a, b, c!"
      */
     public String collectDistinctNames(final List<Person> people) {
-        return null;
+
+
+        String extra="Distinct names:";
+         String distinctName=people.stream()
+                            .map(Person::getName)
+                            .distinct()
+                            .collect(Collectors.joining(", ", " ", "!"));
```

Comment:
Good job! Almost.
I'm glad that you've found one more collector (I was expecting for someone to discover it).
But
You don't need that `String extra = ...` variable. Try to use collector possibilities only.